selectors/views/channel_sidebar.ts

```diff
@@ -55,7 +60,8 @@ export const getChannelsInCategoryOrder = (() => {
         (state: GlobalState) => getChannelsByCategory(state, getCurrentTeamId(state)),
         getCurrentChannel,
         (state: GlobalState) => getUnreadChannelIds(state),
-        (collapsedState, categories, channelsByCategory, currentChannel, unreadChannelIds) => {
+        (state: GlobalState) => state.storage.storage,
```

Comment:
Good points, how do I access and deploy to the test server to check this? I can look around and see if I can come up with an alternative to passing in the whole storage object

Reply:
I did a bit of testing on this. We are already using storage as an argument in `getCollapsedStateForAllCategories`, using it here won't make any change to the performance because it's already being used in this selector.

You are right in saying that it re-renders `SidebarChannelList` every time you type but it was already doing that because of `getCollapsedStateForAllCategories`. From what I tested, it didn't seem to re-render `SidebarCategory`

Reply:
Good find. I forgot that it backed collapsed categories right now since we are planning on moving that to the server at some point.

I also checked, and storage isn't actually updated immediately when the user types in the post textbox, so it doesn't update with every key pressed as I thought, so between both of those, this definitely isn't a problem here.